### PR TITLE
Tests: Re-order tests so that "simple" tests run first

### PR DESCRIPTION
**Pull request by Geod24**
See second commit, it's fairly simple.

### CODE REVIEW

**Review by hewison-chris — APPROVED** [review]
LGTM

**Review by linked0 — APPROVED** [review]
LGTM, one nitpick.

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -211,10 +211,24 @@ private UnitTestResult customModuleUnitTester ()
...
+    struct Tests
+    {
+        /// Simple tests are every tests not in 'agora.tests' package
+        /// They run serially and first, as they might be checking GC usage
+        ModTest[] simple;
+
+        /// Integrations test that should run single-threaded
+        ModTest[] serial;
+        /// Integration tests that run multi-threaded (unless disabled)
+        ModTest[] parallel;
```
Comment (linked0) [inline source/agora/test/Base.d]:
```suggestion
        /// Integrations test that should run single-threaded
        ModTest[] serial;
        
        /// Integration tests that run multi-threaded (unless disabled)
        ModTest[] parallel;
```

**Review by kchulj — APPROVED** [review]
LGTM

**Pull request merged**